src/components/tasks/card/index.tsx

```diff
@@ -242,6 +237,11 @@ const Card: FC<CardProps> = ({
             });
     };
 
+    const onEditRoute = () => {
+        setIsEditMode(true);
+    };
+    shouldEdit = shouldEdit && isUserAuthorized && isEditMode;
```

Comment:
- I think having this boolean variable near `isEditMode` would be better as it will be clear that the of `isEditMode` is used to make a decision.

Reply:
disn't understand your pount here, can you please explain

Reply:
- having the state and it's dependent variables together makes it easier to read and made future edits to the code.
keeping this in mind should we move the block below near `isEditMode` useState? 
```js
shouldEdit = shouldEdit && isUserAuthorized && isEditMode;
```

Reply:
Also `shoulEdit` is props coming from the parent. Props should not be modified by the child components.